File: app/src/release/java/jp/gcreate/product/filteredhatebu/util/ReleaseLogTree.java
package jp.gcreate.product.filteredhatebu.util;

import android.util.Log;

import com.crashlytics.android.Crashlytics;

import timber.log.Timber;

/**
 * Copyright 2016 G-CREATE
 */

public class ReleaseLogTree extends Timber.Tree {
    @Override
    protected void log(int priority, String tag, String message, Throwable t) {
        if ( priority == Log.VERBOSE || priority == Log.DEBUG) {
            return;
        }

        Crashlytics.log(message);
        if (t != null) {
            Crashlytics.log(priority, tag, message);
            Crashlytics.logException(t);
        }
    }
}

File: app/src/release/java/jp/gcreate/product/filteredhatebu/util/ReleaseLogTree.kt
package jp.gcreate.product.filteredhatebu.util

import android.util.Log
import com.google.firebase.crashlytics.FirebaseCrashlytics
import timber.log.Timber

/**
 * Copyright 2016 G-CREATE
 */
class ReleaseLogTree : Timber.Tree() {
    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        if (priority == Log.VERBOSE || priority == Log.DEBUG) {
            return
        }
        FirebaseCrashlytics.getInstance().log(message)
        if (t != null) {
            FirebaseCrashlytics.getInstance().recordException(t)
        }
    }
}